EIPS/eip-4824.md

```diff
@@ -0,0 +1,288 @@
+    function daoURI() external view returns (string _daoURI);
+}
+```
+
+The EIP-4824 DAO JSON-LD Schema mentioned above.
+
+```json
+{
+	"@context": "http://www.daostar.org/schemas",
+	"type": "DAO",
+	"name": "<name of the DAO>",
+	"description": "<description>",
+	"membersURI": "<URI>",
+	"proposalsURI": "<URI>",
+	"activityLogURI": "<URI>",
+	"governanceURI": "<URI>"
+}
+```
+
+A DAO MAY inherit the above interface above or it MAY create an external registration contract that is compliant with this EIP. The external registration contract MUST store the DAO’s primary address.
+
+```solidity
+pragma solidity ^0.8.1;
+
+/// @title EIP-4824 DAOs
+/// @dev See <https://eips.ethereum.org/EIPS/eip-4824>
+
+error NotOwner();
+error NotOffered();
+
+contract EIP4824Registration is EIP4824 {
+
+		string private daoURI;
+		address daoAddress;
+mapping (address => bool) daoRegistered;
+		address offeredAddress;
+
+		event TransferOffered(address currentOwner, address newOwner);
+		event TransferAccepted(address oldOwner, address newOwner);
+
+		function initialize(address _daoAddress) {
+			if (_daoAddress == msg.sender) {
+				daoAddress = _daoAddress;
+				emit TransferAccepted(address(0), _daoAddress);
+			} else {
+				offeredAddress = _daoAddress;
+				emit TransferOffered(address(0), _daoAddress);
+			}
+
+		}
+
+		function daoURI() external view returns (string _daoURI) {
+			return daoURI;
+		}
+		
+    function offerTransfer(address _offeredAddress) external {
+			if (msg.sender != daoAddress) revert NotOwner();
+			offeredAddress = _offeredAddress;
+
+			emit TransferOffered(daoAddress, _offeredAddress);
+		}
+
+    function acceptTransfer() external {
+			address _oldOwner = daoAddress;
+			if (msg.sender != offeredAddress) revert NotOffered();
+			daoAddress = msg.sender;
+			emit TransferAccepted(_oldOwner, daoAddress);
+		}
+}
+```
+
+If a DAO uses an external registration contract, the DAO SHOULD use a common registration factory contract to enable efficient network indexing.
+
+```solidity
+pragma solidity ^0.8.1;
+
+/// @title EIP-4824 DAOs
+/// @dev See <https://eips.ethereum.org/EIPS/eip-4824>
+
+interface EIP4824RegistrationFactory is CloneFactory {
+    function summonRegistration();
+}
+```
+
+### Members
+
+Members JSON-LD Schema.
+
+```json
+{
+  "@context": "<http://www.daostar.org/schemas>",
+  "type": "DAO",
+  "name": "<name of the DAO>",
+  "members": [
+    {
+      "type": "EthereumAddress",
+      "id": "<address or other identifier>"
+    },
+    {
+      "type": "EthereumAddress",
+      "id": "<address or other identifier>"
+    }
+  ]
+}
```

Comment:
```suggestion
{
  "@context": "<http://www.daostar.org/schemas>",
  "type": "DAO",
  "name": "<name of the DAO>",
  "members": [{
    "type": "EthereumAddress",
    "id": "<address or other identifier>"
  }, {
    "type": "EthereumAddress",
    "id": "<address or other identifier>"
  }]
}
```
My personal preference on how JSON should be formatted. This should probably be standardized.